File: src/test/java/input/InputReaderFactoryTest.java
package input;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class InputReaderFactoryTest {

    @Test
    public void testCreateUserInputReader() {
        String[] args = new String[]{};

        var createdInputReader = new InputReaderFactory().createInputReader(args);

        Assertions.assertInstanceOf(UserInputReader.class, createdInputReader);
    }

    @Test
    public void testCreateFileInputReader() {
        String[] args = new String[]{"somefile.txt"};

        var createdInputReader = new InputReaderFactory().createInputReader(args);

        Assertions.assertInstanceOf(FileInputReader.class, createdInputReader);
    }

    @Test
    public void testCreateUserInputReaderOnNullInput() {
        String[] args = null;

        var createdInputReader = new InputReaderFactory().createInputReader(args);

        Assertions.assertInstanceOf(UserInputReader.class, createdInputReader);
    }

    @Test
    public void testCreateUserInputReaderOnInvalidInput() {
        String[] args = new String[]{"these arguments are not valid"};

        var createdInputReader = new InputReaderFactory().createInputReader(args);

        Assertions.assertInstanceOf(UserInputReader.class, createdInputReader);
    }
}

File: src/test/kotlin/at/erste/io/InputReaderFactoryTest.kt
package at.erste.io

import org.junit.jupiter.api.Test
import kotlin.test.assertTrue

class InputReaderFactoryTest {
    val inputReaderFactory = InputReaderFactory()

    @Test
    fun testCreateConsoleReader() {
        val reader = inputReaderFactory.createInputReader(null)
        assertTrue(reader is ConsoleInputReader)
    }

    @Test
    fun testCreateFileReader() {
        val reader = inputReaderFactory.createInputReader("any_file_name.txt")
        assertTrue(reader is FileInputReader)
    }
}
